bin/ed

```diff
@@ -395,6 +396,15 @@ sub escape_line {
     return $s;
 }
 
+# does not modify buffer
+sub edPipe {
+    return unless defined $args[0];
+    my $rc = system $args[0];
```

Comment:
Let's be a lot more careful here. This is `system` in the single argument form, which will have the shell translate the string before running it. Let's either check the string for safety, or convert this to the list form with the braces syntax.

Reply:
> test1: shell pattern: "!for i in 1 2 3; do echo $i; done"
In my understanding, the "!" sub-command in ed is expected to run a shell command, or the above example wouldn't work. 
This works the same in both OpenBSD ed and GNU ed. In terms of being careful, GNU has a restricted mode which is enabled by running "red" or "ed -r". The "!" sub- command is not available in restricted mode. OpenBSD doesn't have this mode; probably because a text editor doesn't need to be ultra secure, and no one will want to run restricted-ed anyway.

Reply:
After thinking about this, I agree. It's what the command does. Thanks for looking at it again. Maybe we can add it later. I've added #409 for this.